packages/twenty-front/src/modules/favorites/components/CurrentWorkspaceMemberFavorites.tsx

```diff
@@ -1,127 +1,229 @@
-              />
-            );
-          })}
-        </>
+    await renameFolder(folder.folderId, value);
+    setIsRenaming(false);
+  };
+
+  const handleDelete = async () => {
+    if (folder.favorites.length > 0) {
+      setIsDeleteModalOpen(true);
+      closeDropdown();
+    } else {
+      await deleteFolder(folder.folderId);
+      closeDropdown();
+    }
+  };
+
+  const handleConfirmDelete = async () => {
+    await deleteFolder(folder.folderId);
+    setIsDeleteModalOpen(false);
+  };
+
+  const rightOptions = (
+    <Dropdown
+      dropdownId={`favorite-folder-${folder.folderId}`}
+      dropdownHotkeyScope={{
+        scope: FavoriteFolderHotkeyScope.FavoriteFolderRightIconDropdown,
+      }}
+      data-select-disable
+      clickableComponent={
+        <IconDotsVertical
+          size={theme.icon.size.md}
+          color={theme.color.gray50}
+        />
+      }
+      dropdownPlacement="right"
+      dropdownOffset={{ y: -15 }}
+      dropdownComponents={
+        <DropdownMenuItemsContainer>
+          <MenuItem
+            LeftIcon={IconPencil}
+            onClick={() => {
+              setIsRenaming(true);
+              closeDropdown();
+            }}
+            accent="default"
+            text="Rename"
+          />
+          <MenuItem
+            LeftIcon={IconTrash}
+            onClick={handleDelete}
+            accent="danger"
+            text="Delete"
+          />
+        </DropdownMenuItemsContainer>
       }
     />
   );
 
   return (
-    <StyledContainer>
-      <NavigationDrawerAnimatedCollapseWrapper>
-        <NavigationDrawerSectionTitle
-          label="Favorites"
-          onClick={() => toggleNavigationSection()}
-        />
-      </NavigationDrawerAnimatedCollapseWrapper>
-
-      {isNavigationSectionOpen && (
-        <ScrollWrapper contextProviderName="navigationDrawer">
-          <NavigationDrawerItemsCollapsedContainer isGroup={isGroup}>
-            {draggableListContent}
-          </NavigationDrawerItemsCollapsedContainer>
-        </ScrollWrapper>
-      )}
-    </StyledContainer>
+    <>
+      <NavigationDrawerItemsCollapsedContainer
+        key={folder.folderId}
+        isGroup={isGroup}
+      >
+        {isRenaming ? (
+          <NavigationDrawerInput
+            Icon={IconFolder}
+            value={folderName}
+            onChange={setFolderName}
+            onSubmit={handleSubmitRename}
+            onCancel={handleCancelRename}
+            onClickOutside={handleClickOutside}
+            hotkeyScope="favorites-folder-input"
+          />
+        ) : (
+          <NavigationDrawerItem
+            key={folder.folderId}
+            label={folder.folderName}
+            Icon={IconFolder}
+            onClick={() => onToggle(folder.folderId)}
+            rightOptions={rightOptions}
+            className="navigation-drawer-item"
+            isDropdownOpen={isDropdownOpen}
+          />
+        )}
+
+        {isOpen && (
+          <DraggableList
```

Comment:
this can be treated in another PR but the DraggableList should be higher so we can Drag favorites between folder, and folder between folders too